src/storage/rdb.cc

```diff
@@ -356,17 +378,28 @@ StatusOr<std::vector<MemberScore>> RDB::LoadZSetWithZipList() {
   return zset;
 }
 
-Status RDB::Restore(const std::string &key, uint64_t ttl_ms) {
+Status RDB::Restore(const std::string &key, std::string_view payload, uint64_t ttl_ms) {
   rocksdb::Status db_status;
 
   // Check the checksum of the payload
-  GET_OR_RET(VerifyPayloadChecksum());
+  GET_OR_RET(VerifyPayloadChecksum(payload));
 
   auto type = GET_OR_RET(LoadObjectType());
+
+  auto value = GET_OR_RET(loadRdbObject(type, key));
+
+  return saveRdbObject(type, key, value, ttl_ms);  // NOLINT
```

Comment:
Why do we need to add NOLINT for this line?

Reply:
The lint report falsely indicates a memory leak, but I can't find any potential for a memory leak

Reply:
ok, it should be if the ASAN in CI is good.